x-pack/legacy/plugins/alerting/server/init.ts

```diff
@@ -17,50 +18,89 @@ import {
 } from './routes';
 import { AlertingPlugin, Services } from './types';
 import { AlertTypeRegistry } from './alert_type_registry';
-import { AlertsClient } from './alerts_client';
+import { AlertsClient, CreateAPIKeyResult } from './alerts_client';
 import { SpacesPlugin } from '../../spaces';
+import { KibanaRequest } from '../../../../../src/core/server';
+import { PluginSetupContract as SecurityPluginSetupContract } from '../../../../plugins/security/server';
 import { createOptionalPlugin } from '../../../server/lib/optional_plugin';
 
 export function init(server: Legacy.Server) {
   const config = server.config();
+  const kbnServer = (server as unknown) as KbnServer;
+  const taskManager = server.plugins.task_manager!;
+  const { callWithRequest } = server.plugins.elasticsearch.getCluster('admin');
   const spaces = createOptionalPlugin<SpacesPlugin>(
     config,
     'xpack.spaces',
     server.plugins,
     'spaces'
   );
-
-  const { callWithInternalUser } = server.plugins.elasticsearch.getCluster('admin');
-  const savedObjectsRepositoryWithInternalUser = server.savedObjects.getSavedObjectsRepository(
-    callWithInternalUser
+  const security = createOptionalPlugin<SecurityPluginSetupContract>(
+    config,
+    'xpack.security',
+    kbnServer.newPlatform.setup.plugins,
+    'security'
   );
 
-  function getServices(basePath: string): Services {
-    const fakeRequest: any = {
-      headers: {},
-      getBasePath: () => basePath,
-    };
+  server.plugins.xpack_main.registerFeature({
+    id: 'alerting',
+    name: 'Alerting',
+    app: ['alerting', 'kibana'],
+    privileges: {
+      all: {
```

Comment:
Same comment as above about restricting access to APIs: https://github.com/elastic/kibana/blob/2b4f2a259e02f2a788d374a88b036c0444c44ed3/x-pack/legacy/plugins/xpack_main/server/lib/feature_registry/feature_registry.ts#L33-L57